### PR TITLE
[frontend] Remove unnecessary if-clause in PUT

### PR DESCRIPTION
**Pull request by snalli**
Remove unnecessary if-clause in PutOperation

### CODE REVIEW

File: ambry-router/src/main/java/com/github/ambry/router/PutOperation.java [hunk 1]
```diff
@@ -1184,21 +1184,11 @@ private void prepareForBuilding(int chunkIndex) {
...
-        } else if (chunkIndex == 0) {
-          // TODO: need more data to distinguish between Simple and Data Chunk
-          blobDataType = BlobDataType.DATACHUNK;
-        } else {
-          blobDataType = BlobDataType.DATACHUNK;
-        }
+        BlobDataType blobDataType = isMetadataChunk() ? BlobDataType.METADATA : BlobDataType.DATACHUNK;
 
-        // if this is part of a retry, make sure no previously attempted partitions are retried.
-        if (partitionId != null) {
```
Comment (ankagrawal) [inline ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:1198]: Why remove this check and add to `attemptedPartitionIds`?

Comment (snalli) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:1198]: code with less if-clauses is just flows and easier to read.
The old code and this one are equivalent.
`attemptedPartitionIds` is not used anywhere else.
In this new code, if we fail and retry, we already have recorded the prev partition and will exclude it. Same as old code. Although if we succeed, the size of attemptedPartitionIds will be 1 in this case and 0 in old code.

Comment (snalli) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:1198]: Not neccesary to merge but something i noticed while reading the code today.

Comment (ankagrawal) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:1198]: I must be missing something -- Why do you say that `attemptedPartitionIds` is not used anywhere? Its used in the immediately next line after the line you deleted.

> we already have recorded the prev partition and will exclude it

How and where have we recorded the previous partition and where's the code to exclude it?

Comment (ankagrawal) [reply ambry-router/src/main/java/com/github/ambry/router/PutOperation.java:1198]: I see that you added the `attemptedPartitionIds` in the next line.

**Review by ankagrawal — APPROVED** [review]

**Review by ankagrawal — APPROVED** [review]

**Review by Arun-LinkedIn — APPROVED** [review]

**Pull request merged**